draft-ietf-tls-tls13.md

```diff
@@ -3947,23 +3988,27 @@ TLS-compliant application MUST implement the following TLS extensions:
   * Negotiated Groups ("supported_groups"; {{negotiated-groups}})
   * Key Share ("key_share"; {{key-share}})
   * Pre-Shared Key ("pre_shared_key"; {{pre-shared-key-extension}})
+  * Pre-Shared Key Exchange Modes ("psk_key_exchange_modes";
+    {{pre-shared-key-exchange-modes}})
   * Server Name Indication ("server_name"; Section 3 of {{RFC6066}})
   * Cookie ("cookie"; {{cookie}})
 
 All implementations MUST send and use these extensions when offering
-applicable cipher suites:
+the relevant parameters:
 
   * "supported_versions" is REQUIRED for all ClientHello messages.
-  * "signature_algorithms" is REQUIRED for certificate authenticated cipher suites.
-  * "supported_groups" and "key_share" are REQUIRED for DHE or ECDHE cipher suites.
-  * "pre_shared_key" is REQUIRED for PSK cipher suites.
-  * "cookie" is REQUIRED for all cipher suites.
-
-When negotiating use of applicable cipher suites, endpoints MUST abort the
-handshake with a "missing_extension" alert if the required extension was
-not provided. Any endpoint that receives an invalid combination of cipher
-suites and extensions MAY abort the connection with a "missing_extension"
-alert, regardless of negotiated parameters.
+  * "signature_algorithms" is REQUIRED for certificate authentication.
+  * "supported_groups" and "key_share" are REQUIRED for DHE or ECDHE key
+    establishment.
+  * "pre_shared_key" is REQUIRED for PSK key establishment.
+  * "psk_key_exchange_modes" is required when PSKs are offered.
```

Comment:
I'm not 100% clear on whether the server is required to send this.  It seems not.  But you missed it from the IANA considerations table, so I can't verify.